oap-server/analyzer/meter-analyzer/src/main/java/org/apache/skywalking/oap/meter/analyzer/k8s/K8sInfoRegistry.java

```diff
@@ -96,6 +98,8 @@ private K8sInfoRegistry() {
                     .filter(it -> it.getMetadata() != null)
                     .filter(it -> it.getSpec() != null)
                     .filter(it -> requireNonNull(it.getSpec()).getSelector() != null)
+                    .filter(it -> it.getMetadata().getNamespace() != null
+                            && it.getMetadata().getNamespace().equals(pod.get().getMetadata().getNamespace()))
```

Comment:
This should move after the next `filter` because it is also for checking `spec#selector`.

Reply:
Do you mean so?
![image](https://user-images.githubusercontent.com/37059529/226537364-b6e0c131-5daf-4c56-a359-b7b7677e72b1.png)

Reply:
Do L103 - L106 first.

Reply:
I think your codes are correct, I just want to keep logic more clear.

Reply:
I think to filter the namespace first, the data hit by the next filter will be less. :)

Reply:
Then you should move before `it -> requireNonNull(it.getSpec()).getSelector() != null`.

Reply:
got it